File: apps/docs/src/prework/examples/procedural.ts
export type Request = Readonly<{
  equipmentChecked: boolean;
  flareAlert: boolean;
}>;
export type Reason = "EquipmentNotChecked" | "FlareAlertActive";

export function check(request: Request): Reason[] {
  const reasons: Reason[] = [];
  if (!request.equipmentChecked) reasons.push("EquipmentNotChecked");
  if (request.flareAlert) reasons.push("FlareAlertActive");
  return reasons;
}

// check({ equipmentChecked: false, flareAlert: true })
// => ["EquipmentNotChecked", "FlareAlertActive"]

File: apps/docs/src/prework/examples/python/procedural.py
from dataclasses import dataclass
from typing import Literal


@dataclass(frozen=True)
class Request:
    equipment_checked: bool
    flare_alert: bool


Reason = Literal["EquipmentNotChecked", "FlareAlertActive"]


def check(request: Request) -> list[Reason]:
    reasons: list[Reason] = []
    if not request.equipment_checked:
        reasons.append("EquipmentNotChecked")
    if request.flare_alert:
        reasons.append("FlareAlertActive")
    return reasons
